transform/models/intermediate/imputation/int_imputation__five_minute_aggregation.sql

```diff
@@ -0,0 +1,204 @@
+{{ config(materialized='table') }}
+
+with unimputed as (
+    select
+        id,
+        lane,
+        sample_date,
+        sample_timestamp,
+        volume_sum,
+        occupancy_avg,
+        speed_weighted,
+        extract(hour from sample_timestamp) as sample_hour,
+        coalesce(speed_weighted, (volume_sum * 22) / nullifzero(occupancy_avg) * (1 / 5280) * 12)
+            as speed_five_mins
+    from {{ ref('int_clearinghouse__five_minute_station_agg') }}
+    where sample_date = dateadd(day, -3, current_date)
+),
+
+-- classify the data that needs imputation
+counts_with_imputation_status as (
+    select
+        unimputed.*,
+        case
+            -- when volume and occupancy is present we already used the formula to calculate the speed
+            -- here we will impute where all three has null value
+            -- where either volume or occupany only null along with null speed
+            -- however, we will not impute where volime zero, occupancy zero and speed is null
+            when
+                (unimputed.volume_sum is null and unimputed.occupancy_avg is null and unimputed.speed_five_mins is null)
+                or (
+                    unimputed.volume_sum is not null
+                    and unimputed.occupancy_avg is null
+                    and unimputed.speed_five_mins is null
+                )
+                or (
+                    unimputed.volume_sum is null
+                    and unimputed.occupancy_avg is not null
+                    and unimputed.speed_five_mins is null
+                )
+                then 'yes'
+            else 'no'
+        end as is_imputation_required
```

Comment:
I suspect this can be simplified a lot (to the point where a `case` statement is not necessary). If the goal is to impute when

1. All three are null
2. Either volume or occupancy is null

I think it will be covered by `(volume is null) or (occupancy is null)`, and having a boolean dtype. You have another case where you don't impute when occupancy is null, volume is null, and speed is _not_ null. But I suspect you might want to impute in that case as well.

In any event, I do thing doing boolean logic here would make things much simpler than a big case statement:
```suggestion
        (unimputed.volume_sum is null) or (unimputed.occupancy_avg is null) as imputation_required
```

Reply:
@ian-r-rose , sure, go ahead and replace this with enhanced code! Thank you!